onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -0,0 +1,185 @@
+    queryset = OpenData.objects.filter()
+    lookup_field = 'uuid'
+    serializer_class = OpenDataSerializer
+    flattened_dict = {}
+    MAX_INSTANCES_PER_REQUEST = 1000
+
+    def get_tableau_type(self, xform_type):
+        '''
+        Returns a tableau-supported type based on a xform type.
+        '''
+        tableau_types = {
+            'integer': 'int',
+            'decimal': 'float',
+            'dateTime': 'datetime',
+            'text': 'string'
+        }
+
+        return tableau_types.get(xform_type, 'string')
+
+    def flatten_xform_columns(self, json_of_columns_fields):
+        '''
+        Flattens a json of column fields and the result is set to a class
+        variable.
+        '''
+        for a in json_of_columns_fields:
+            self.flattened_dict[a.get('name')] = self.get_tableau_type(
+                a.get('type')
+            )
+            # using IGNORED_FIELD_TYPES so that choice values are not included.
+            if a.get('children') and a.get('type') not in IGNORED_FIELD_TYPES:
+                self.flatten_xform_columns(a.get('children'))
+
+    def get_tableau_column_headers(self):
+        '''
+        Retrieve columns headers that are valid in tableau.
+        '''
+        tableau_colulmn_headers = []
+
+        def append_to_tableau_colulmn_headers(header, question_type=None):
+            quest_type = 'string'
+            if question_type:
+                quest_type = question_type
+
+            # alias can be updated in the future to question labels
+            tableau_colulmn_headers.append({
+                'id': header,
+                'dataType': quest_type,
+                'alias': header
+            })
+
+        # using nested loops to determine what valid data types to set for
+        # tableau.
+        for header in self.xform_headers:
+            for quest_name, quest_type in self.flattened_dict.items():
+                if header == quest_name or header.endswith('_%s' % quest_name):
+                    append_to_tableau_colulmn_headers(header, quest_type)
+                    break
+            else:
+                append_to_tableau_colulmn_headers(header)
+
+        return tableau_colulmn_headers
+
+    @detail_route(methods=['GET'])
+    def data(self, request, **kwargs):
+        self.object = self.get_object()
+        # get greater than value and cast it to an int
+        gt = request.query_params.get('gt_id')
+        gt = gt and parse_int(gt)
+
+        data = []
+        if isinstance(self.object.content_object, XForm):
+            if not self.object.active:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+
+            xform = self.object.content_object
+            qs_kwargs = {'xform': xform}
+            if gt:
+                qs_kwargs.update({'id__gt': gt})
+
+            instances = Instance.objects.filter(
+                **qs_kwargs
+            )[:self.MAX_INSTANCES_PER_REQUEST]
```

Comment:
I am opposed to this approach, it does not make much sense to me. Please have a look at http://www.django-rest-framework.org/api-guide/pagination/. I can work on this with you when I get back to the office.